File: media-service/src/main/java/org/burgas/mediaservice/mapper/Mapper.java
package org.burgas.mediaservice.mapper;

import org.burgas.mediaservice.dao.Dao;
import org.burgas.mediaservice.dto.Response;
import org.springframework.web.multipart.MultipartFile;

public interface Mapper<D extends Dao, R extends Response> {

    D toDao(final MultipartFile multipartFile);

    R toResponse(D dao);
}

File: course-service/src/main/kotlin/org/burgas/courseservice/mapper/contract/Mapper.kt
package org.burgas.courseservice.mapper.contract

import org.burgas.courseservice.dao.Dao
import org.burgas.courseservice.dto.Dependency
import org.burgas.courseservice.dto.Request
import org.burgas.courseservice.dto.Response

interface Mapper<Req : Request, Ent : Dao, Dep : Dependency, Res : Response> {

    fun toEntity(request: Req): Ent

    fun toDependency(entity: Ent): Dep

    fun toResponse(entity: Ent): Res
}